Description: Skip beautify
Forwarded- not-needed

--- a/scripts/compile-dots.js
+++ b/scripts/compile-dots.js
@@ -4,8 +4,8 @@
 var glob = require('glob')
   , fs = require('fs')
   , path = require('path')
-  , doT = require('dot')
-  , beautify = require('js-beautify').js_beautify;
+  , doT = require('dot');
+//  , beautify = require('js-beautify').js_beautify;
 
 var defsRootPath = process.argv[2] || path.join(__dirname, '../lib');
 
@@ -47,7 +47,7 @@
   removeAlwaysFalsyInOr();
   VARS.forEach(removeUnusedVar);
   code = "'use strict';\nmodule.exports = " + code;
-  code = beautify(code, { indent_size: 2 }) + '\n';
+//  code = beautify(code, { indent_size: 2 }) + '\n';
   fs.writeFileSync(targetPath, code);
   console.log('compiled', keyword);
 
